salalib/point.h

```diff
@@ -60,11 +60,41 @@ class Point {
    int m_processflag;
 public:
    Point()
-      { m_state = EMPTY; m_block = 0; m_misc = 0; m_grid_connections = 0; m_node = NULL; m_processflag = 0; m_merge = NoPixel; m_user_data = NULL; }
+      { m_state = EMPTY; m_block = 0; m_misc = 0; m_grid_connections = 0; m_node = nullptr; m_processflag = 0; m_merge = NoPixel; m_user_data = NULL; }
    Point& operator = (const Point& p)
-      { throw 1; }
+   {
+       m_block = p.m_block;
+       m_state = p.m_state;
+       m_misc = p.m_misc;
+       m_grid_connections = p.m_grid_connections;
+       m_node = p.m_node ? std::unique_ptr<Node>(new Node(*p.m_node)) : nullptr;
```

Comment:
I think a move constructor/assignment operator would be useful here where we just pass ownership of the unique_ptr and null out the original.